internal/github/github.go

```diff
@@ -108,3 +109,57 @@ func (c *Client) GetRawContent(ctx context.Context, path, ref string) ([]byte, e
 	defer body.Close()
 	return io.ReadAll(body)
 }
+
+// CreatePullRequest creates a pull request in the remote repo.
+// At the moment this requires a single remote to be configured,
+// which must have a GitHub HTTPS URL. We assume a base branch of "main".
+func (c *Client) CreatePullRequest(ctx context.Context, repo *Repository, remoteBranch, title, body string) (*PullRequestMetadata, error) {
+	if body == "" {
+		body = "Regenerated all changed APIs. See individual commits for details."
+	}
+	newPR := &github.NewPullRequest{
+		Title:               &title,
+		Head:                &remoteBranch,
+		Base:                github.Ptr("main"),
+		Body:                github.Ptr(body),
+		MaintainerCanModify: github.Ptr(true),
+	}
+	pr, _, err := c.PullRequests.Create(ctx, repo.Owner, repo.Name, newPR)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("PR created: %s\n", pr.GetHTMLURL())
+	pullRequestMetadata := &PullRequestMetadata{Repo: repo, Number: pr.GetNumber()}
+	return pullRequestMetadata, nil
+}
+
+// GetGitHubRepoFromRemote parses the GitHub repo name from the remote for this repository.
+// There must only be a single remote with a GitHub URL (as the first URL), in order to provide an
+// unambiguous result.
+// Remotes without any URLs, or where the first URL does not start with https://github.com/ are ignored.
+func GetGitHubRepoFromRemote(repo *gitrepo.Repository) (*Repository, error) {
+	remotes, err := repo.Remotes()
+	if err != nil {
+		return nil, err
+	}
+	gitHubRemoteNames := []string{}
+	gitHubUrl := ""
+	for _, remote := range remotes {
+		urls := remote.Config().URLs
+		if len(urls) > 0 && strings.HasPrefix(urls[0], "https://github.com/") {
+			gitHubRemoteNames = append(gitHubRemoteNames, remote.Config().Name)
+			gitHubUrl = urls[0]
+		}
+	}
+
+	if len(gitHubRemoteNames) == 0 {
+		return nil, fmt.Errorf("no GitHub remotes found")
+	}
+
+	if len(gitHubRemoteNames) > 1 {
+		joinedRemoteNames := strings.Join(gitHubRemoteNames, ", ")
+		return nil, fmt.Errorf("can only determine the GitHub repo with a single matching remote; GitHub remotes in repo: %s", joinedRemoteNames)
```

Comment:
Inline the strings.Join, or at least give the variable a much shorter name (a single letter is fine).